stl/inc/shared_mutex

```diff
@@ -391,6 +401,7 @@ public:
 
     template <class _Rep, class _Period>
     _NODISCARD_TRY_CHANGE_STATE
+    _Requires_lock_not_held_(*this->_Pmtx)

```

Comment:
Is this necessary?

Reply:
According to Standard calling it with lock held is Undefined Behaviour.

Reply:
I mean, given that `_Acquires_nonreentrant_mutex_` below, I don't think it's necessary since `_Acquires_nonreentrant_mutex_` already implies `_Requires_lock_not_held_`

Reply:
This is a precondition.
Post condition is lock held only when returns true.